single_kernel_mongo/managers/mongodb_operator.py

```diff
@@ -995,19 +1003,26 @@ def is_removing_last_replica(self) -> bool:
     def basic_statuses(self) -> list[StatusObject]:
         """Basic checks."""
         statuses = []
-        if not self.cluster_manager.is_valid_mongos_integration():
-            statuses.append(MongoDBStatuses.INVALID_MONGOS_REL.value)
-
         if not self.backup_manager.is_valid_s3_integration():
             statuses.append(MongoDBStatuses.INVALID_S3_REL.value)
-
-        if self.state.client_relations and self.state.is_sharding_component:
-            statuses.append(MongoDBStatuses.INVALID_DB_REL.value)
+        # Add valid statuses for all invalid relations integratedd
+        for relation_name in [
+            RelationNames.DATABASE,
+            RelationNames.SHARDING,
+            RelationNames.CONFIG_SERVER,
+            RelationNames.CLUSTER,
+        ]:
+            if (
+                self.model.relations[relation_name.value]
+                and (status := self.get_relation_feasible_status(relation_name)) is not None
+            ):
+                statuses.append(status)
 
         if not self.state.is_sharding_component and self.state.has_sharding_integration:
-            statuses.append(MongoDBStatuses.INVALID_SHARDING_REL.value)
-        elif rev_status := self.cluster_version_checker.get_cluster_mismatched_revision_status():
             # don't bother checking revision mismatch on sharding interface if replica
```

Comment:
should this go?

Reply:
It's been moved below with the same comment.
There's no need to check for invalid versions on invalid relations so we early return and otherwise continue with the cluster status checking.